config/config.go

```diff
@@ -195,6 +196,10 @@ func LoadConfig(fileLocation string, vip *viper.Viper) (err error) {
 	err = vip.ReadInConfig()
```

Comment:
just do `return vip.ReadInConfig()` here. the upstream caller does log.fatal() on error already.

in essence, we simply don't fall back to do default config.

Reply:
@countvonzero yeah, but the problem is that this logic is needed in tests, and we pretty much somehow assume kinda everywhere that this is tried to be ALWAYS read.

That logic is fatally flawed and should be reworked completely.

Nevertheless, that fix is meant to fix the problem faced in the real production application now.

Reply:
i refactored to return fatal error on any failure to load specified config file.
if config file is not specified we will try to load default config file if it exists